Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1339,14 +1358,14 @@ ExceptionOr<void> HTMLElement::showPopover()
 
     popoverData()->setPreviouslyFocusedElement(nullptr);
 
-    // FIXME: Run popover focusing steps.
-
     Style::PseudoClassChangeInvalidation styleInvalidation(*this, {
         { CSSSelector::PseudoClassOpen, true },
         { CSSSelector::PseudoClassClosed, false }
     });
     popoverData()->setVisibilityState(PopoverVisibilityState::Showing);
```

Comment:
If the popover focusing steps fail should we be setting the visibility state still to showing?

Reply:
The focusing steps should run after the setting the visibility state to showing (the FIXME comment was wrongly placed). See https://html.spec.whatwg.org/#show-popover

Reply:
Ah, you are right; just did a read through of that piece of the spec this morning.